packages/kbn-securitysolution-utils/src/path_validations/index.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+export const FILENAME_WILDCARD_WARNING = i18n.translate('utils.filename.wildcardWarning', {
+  defaultMessage: `A wildcard in the filename will affect the endpoint's performance`,
+});
+
+export const FILEPATH_WARNING = i18n.translate('utils.filename.pathWarning', {
+  defaultMessage: `Path may be formed incorrectly; verify value`,
+});
+
+export enum ConditionEntryField {
+  HASH = 'process.hash.*',
```

Comment:
I'm not sure we should be using this value for `hash` since its not actually valid. We used this in Trusted APps initially so that we did not have the user pick `md5`, `sha1` or `sha256`.

I think you copied most of this from the Trusted apps work we did, but much of that is still pointing to the prior implementation where it did use the `ExceptionListItemSchema` types or follow much of the same patterns as other artifacts that have come after trusted apps.

Also - why are we adding trusted apps specific path validations to `@kbn/securitysolution-utils`? they should be more generic here, right?

Reply:
> I'm not sure we should be using this value for `hash` since its not actually valid. We used this in Trusted APps initially so that we did not have the user pick `md5`, `sha1` or `sha256`.

I see. 

> I think you copied most of this from the Trusted apps work we did, but much of that is still pointing to the prior implementation where it did use the `ExceptionListItemSchema` types or follow much of the same patterns as other artifacts that have come after trusted apps.

Yeah I moved the code that was [here](https://github.com/elastic/kibana/blob/a8fb87467dd8cdb56531797ec98e934bcd1b6048/x-pack/plugins/security_solution/common/endpoint/types/trusted_apps.ts#L70)

> Also - why are we adding trusted apps specific path validations to `@kbn/securitysolution-utils`? they should be more generic here, right?

I thought this would be the only aptly named package to move the code that validates paths and shows soft warnings for both TAs and EFs without having to do a lot.

I also thought about creating a new package and moving all the code in there. That seemed more involved with deciding a proper package name (maybe `kbn-securitysolution-path-utils`) and carefully adding it to all the config files to make sure that it was built correctly. Nonetheless, I'm open to suggestions for a more generic solution.

Reply:
I think I'm fine if there's a few domain specifics such as trusted applications in this package. Over time if it grows or if you want to split it out I think that's fine.

For the most part I don't think we have seen other people/domains/packages wanting to use exception lists such as for o11y (observability rules) but we should think if they did want to use it one day how much work would that potentially be.

But again if it feels cleaner to split this out into its own package I think that's fine as well. Either way I think I'm good here. 👍